server/sonar-server/src/test/java/org/sonar/server/plugins/ws/PluginsWsMediumTest.java

```diff
@@ -0,0 +1,171 @@
+
+    // 2 - login as admin and install one plugin, update another, verify pending status in the process
+    userSessionRule.login().setGlobalPermissions(GlobalPermissions.SYSTEM_ADMIN);
+    wsTester.newPostRequest("api/plugins", "update").setParam("key", "decoy").execute().assertNoContent();
+
+    wsTester.newGetRequest("api/plugins", "pending").execute().assertJson("{" +
+        "  \"installing\": [" +
+        "    {" +
+        "      \"key\": \"decoy\"," +
+        "      \"version\": \"1.1\"" +
+        "    }" +
+        "  ]," +
+        "  \"removing\": []" +
+        "}");
+
+    wsTester.newPostRequest("api/plugins", "install").setParam("key", "foo").execute().assertNoContent();
+
+    wsTester.newGetRequest("api/plugins", "pending").execute().assertJson("{" +
+        "  \"installing\": [" +
+        "    {" +
+        "      \"key\": \"decoy\"," +
+        "      \"version\": \"1.1\"" +
+        "    }," +
+        "    {" +
+        "      \"key\": \"foo\"," +
+        "      \"version\": \"1.0\"" +
+        "    }" +
+        "  ]," +
+        "  \"removing\": []" +
+        "}");
+
+    // 3 - simulate SQ restart
+    wsTester = restartServerTester();
+
+    // 4 - make sure plugin is installed
+    wsTester.newGetRequest("api/plugins", "installed").execute().assertJson("{" +
+      "  \"plugins\": [" +
+      "    {" +
+      "      \"key\": \"decoy\"," +
+      "      \"version\": \"1.1\"" +
+      "    }," +
+      "    {" +
+      "      \"key\": \"foo\"," +
+      "      \"version\": \"1.0\"" +
+      "    }" +
+      "  ]" +
+      "}"
+      );
+
+    // 5 - login as admin again and uninstall a plugin, verify pending status in the process
+    userSessionRule.login().setGlobalPermissions(GlobalPermissions.SYSTEM_ADMIN);
+    wsTester.newPostRequest("api/plugins", "uninstall").setParam("key", "foo").execute().assertNoContent();
+
+    wsTester.newGetRequest("api/plugins", "pending").execute().assertJson("{" +
+        "  \"installing\": []," +
+        "  \"removing\": [" +
+        "    {" +
+        "      \"key\": \"foo\"," +
+        "      \"version\": \"1.0\"" +
+        "    }" +
+        "  ]," +
+        "}");
+
+    // 6 - simulate SQ restart again
+    wsTester = restartServerTester();
+
+    // 7 - make sure plugin has been uninstalled
+    wsTester.newGetRequest("api/plugins", "installed").execute().assertJson("{" +
+            "  \"plugins\": [" +
+            "    {" +
+            "      \"key\": \"decoy\"," +
+            "      \"version\": \"1.1\"" +
+            "    }" +
+            "  ]" +
+            "}"
+    );
+  }
+
+  private WsTester restartServerTester() {
+    WsTester wsTester;
+    serverTester.restart();
+    // correctly simulate a server restart (ie. user is disconnected)
+    userSessionRule.anonymous();
+    // must use a new WsTester to reference the right PluginWs instance
+    wsTester = new WsTester(serverTester.get(PluginsWs.class));
+    return wsTester;
```

Comment:
I would inline the instanciation and remove the declaration of the intermediate variable.